hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/action/compact/TestAsyncCompaction.java

```diff
@@ -50,7 +51,9 @@
 public class TestAsyncCompaction extends CompactionTestBase {
 
   private HoodieWriteConfig getConfig(Boolean autoCommit) {
-    return getConfigBuilder(autoCommit).build();
+    return getConfigBuilder(autoCommit)
+        .withMetadataConfig(HoodieMetadataConfig.newBuilder().enable(true).validate(true).build())
```

Comment:
Do we want this test suite to run against only metadata enabled table from now on? If we want coverage for both can we parameterize tests?

Reply:
ideally yes. but adds to the runtime. its okay I think.

Reply:
I have also turned on validation. So it must cover both cases, that way.